pennylane/measurements/mid_measure.py

```diff
@@ -218,6 +232,18 @@ def _merge(self, other: "MeasurementValue"):
         merged_measurement_ids = list(set(self.measurement_ids).union(set(other.measurement_ids)))
         merged_measurement_ids.sort()
 
+        merged_measurements = []
+        for item in self.measurements + other.measurements:
+            is_duplicate = False
+            for unique_item in merged_measurements:
+                if item is unique_item:
+                    is_duplicate = True
+
+            if not is_duplicate:
+                merged_measurements.append(item)
+
+        merged_measurements.sort(key=lambda m: m.id)
+
         # create a new function that selects the correct indices for each sub function
         def merged_fn(*x):
             sub_args_1 = (
```

Comment:
Could the line below be `[merged_measurements.index(m) for m in self.measurements]` instead (and correspondingly in line 255)? Then it seems we could drop the references to `measurement_ids` in this function.